ow_faults_detection/src/FaultDetector.cpp

```diff
@@ -0,0 +1,256 @@
+
+  // topics for JPL msgs: system fault messages, see Faults.msg, Arm.msg, Power.msg, PTFaults.msg
+  m_arm_fault_msg_pub = node_handle.advertise<ow_faults_detection::ArmFaults>("/faults/arm_faults_status", 10);
+  m_antenna_fault_msg_pub = node_handle.advertise<ow_faults_detection::PTFaults>("/faults/pt_faults_status", 10);
+  m_camera_fault_msg_pub = node_handle.advertise<ow_faults_detection::CamFaults>("/faults/cam_faults_status", 10);
+  m_power_fault_msg_pub = node_handle.advertise<ow_faults_detection::PowerFaults>("/faults/power_faults_status", 10);
+  m_system_fault_msg_pub = node_handle.advertise<ow_faults_detection::SystemFaults>("/faults/system_faults_status", 10);
+
+}
+
+// Creating Fault Messages
+template<typename fault_msg>
+void FaultDetector::setFaultsMessageHeader(fault_msg& msg){
+  msg.header.stamp = ros::Time::now();
+  msg.header.frame_id = "world";
+}
+
+template<typename bitsetFaultsMsg, typename bitmask>
+void FaultDetector::setBitsetFaultsMessage(bitsetFaultsMsg& msg, bitmask bm) {
+  setFaultsMessageHeader(msg);
+  msg.value = bm.to_ullong();
+}
+
+template<typename fault_msg>
+void FaultDetector::setComponentFaultsMessage(fault_msg& msg, ComponentFaults value) {
+  setFaultsMessageHeader(msg);
+  msg.value = static_cast<uint>(value);
+}
+
+// publish system messages
+void FaultDetector::publishSystemFaultsMessage(){
+  ow_faults_detection::SystemFaults system_faults_msg;
+  setBitsetFaultsMessage(system_faults_msg, m_system_faults_bitset);
+  m_system_fault_msg_pub.publish(system_faults_msg);
+}
+
+//// Publish Camera Messages
+void FaultDetector::cameraTriggerPublishCb(const ros::TimerEvent& t){
+  ow_faults_detection::CamFaults camera_faults_msg;
+  auto diff = m_cam_raw_time - m_cam_trigger_time;
+  if (m_cam_trigger_time <= m_cam_raw_time  
+    &&  m_cam_raw_time <= m_cam_trigger_time + ros::Duration(2) 
+    || diff < ros::Duration(0) && ros::Duration(-1) < diff) {
+    m_system_faults_bitset &= ~isCamExecutionError;
+  } else {
+    m_system_faults_bitset |= isCamExecutionError;
+    setComponentFaultsMessage(camera_faults_msg, ComponentFaults::Hardware);
+  }
+
+  publishSystemFaultsMessage();
+  m_camera_fault_msg_pub.publish(camera_faults_msg);
+}
+
+//// Publish Power Faults Messages
+void FaultDetector::publishPowerSystemFault(){
+  ow_faults_detection::PowerFaults power_faults_msg;
+  //update if fault
+  if (m_temperature_fault || m_soc_fault) {
+    //system
+    m_system_faults_bitset |= isPowerSystemFault;
+    //power
+    setComponentFaultsMessage(power_faults_msg, ComponentFaults::Hardware);
+  } else {
+    m_system_faults_bitset &= ~isPowerSystemFault;
+  }
+  publishSystemFaultsMessage();
+  m_power_fault_msg_pub.publish(power_faults_msg);
+}
+
+// Listeners
+// Arm listeners
+bool FaultDetector::isFlagSet(uint joint, const std::vector<double>& flags) {
+  unsigned int index;
+  auto found = findJointIndex(joint, index);
+
+  if (!found)
+    return false;
```

Comment:
Put brackets around this statement for consistency, and as a generally good coding style for multi-line IF statements.